src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.Cache.cs

```diff
@@ -35,10 +35,10 @@ public abstract partial class JsonTypeInfo
 
         // All of the serializable parameters on a POCO constructor keyed on parameter name.
         // Only parameters which bind to properties are cached.
-        internal JsonPropertyDictionary<JsonParameterInfo>? ParameterCache;
+        internal JsonPropertyDictionary<JsonParameterInfo>? ParameterCache { get; private set; }
 
         // All of the serializable properties on a POCO (except the optional extension property) keyed on property name.
-        internal JsonPropertyDictionary<JsonPropertyInfo>? PropertyCache;
+        internal JsonPropertyDictionary<JsonPropertyInfo>? PropertyCache { get; private protected set; }
```

Comment:
aren't you changing this only in JsonTypeInfo now? I think it could be `private set` - or are we still initializing this in LateAdd?

Reply:
IIRC it's being set by LateAdd. I guess we could refactor it to a factory that returns the cache value to restrict access to the base class only